tests/assoc_sim.R

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/Rscript
+#
+# Copyright 2015 Google Inc. All rights reserved.
+# 
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# 
+#     http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+library("optparse")
+
+# First parse args
+options(stringsAsFactors = FALSE)
+if(!interactive()) {
+  option_list <- list(
+    # Flags
+    make_option(c("--candidates", "-c"), default = "candidates.csv",
+                help = "Filename for candidates"),
+    make_option(c("--params", "-p"), default = "params.csv",
+                help = "Filename for parameters"),
+    make_option(c("--reports", "-r"), default = "reports.csv",
+                help = "Filename for reports"),
+    make_option(c("--map", "-m"), default = "map",
+                help = "Filename *prefix* for map(s)"),
+    make_option(c("--num", "-n"), default = 1e05,
+                help = "Number of reports"),
+    make_option(c("--unif", "-u"), default = FALSE,
+                help = "Run simulation with uniform distribution")
+  )
+  opts <- parse_args(OptionParser(option_list = option_list))
+}    
+
+source("../analysis/R/encode.R")
+source("../analysis/R/decode.R")
+source("../analysis/R/simulation.R")
+source("../analysis/R/read_input.R")
+source("../analysis/R/association.R")
+
+# Read candidates from a csv file
+# Inputs: filename. The file is expected to contain two rows of strings
```

Comment:
I just tried it, and it sort of works, except one var is the wrong length.  Your solution has the same problem I believe -- the shorter vectors end up filled with "NA", and are the wrong length.

Another solution is to have multiple files -- one for each var.  I think that is the simplest thing.

We can talk about this -- I also think the params file should have the var names and types, because it will be too easy to associate the wrong params with the wrong reports, when you have 3 dimensions or more.  That will probably make the analysis blow up and cause us problems.

The input checking on the existing scripts is already quite bad, which has caused us problems, so I want to make these nicer.

$ cat test.csv 
var1,var2
a,1
b,2
 ,3
 ,4


> r=read.csv('test.csv')
> r$var1
[1] a b    
Levels:   a b
> r$var2
[1] 1 2 3 4
> 
> r
  var1 var2
1    a    1
2    b    2
3         3
4         4
> length(r$var1)                                                                                                       
[1] 4
> length(r$var2)
[1] 4